src/app/themes.tsx

```diff
@@ -30,19 +70,31 @@ const StyleManager: React.FC<Props> = ({ children }) => {
         return <>{styles}</>;
     });
 
+    const [chosenTheme, setChosenTheme] = useState(theme);
+
     if (typeof window !== "undefined") {
-        return <>{children}</>;
+        return (
```

Comment:
can these returns be combined? there's a lot of duplicate code between them. Maybe something like:
```tsx
const childrenWithStyleManager = (typeof window !== "undefined") ? children : (
    <StyleSheetManager ...> {children} </...>
);
return // just one return using above variable
```